internal/http/client.go

```diff
@@ -2,24 +2,31 @@ package http
 
 import (
 	"context"
+	"fmt"
 	"io"
 	"io/ioutil"
+	"math/rand"
 	"net/http"
+	"strconv"
+	"time"
 
 	"golang.org/x/time/rate"
 )
 
 // Client is a HTTP client
 type Client struct {
-	*http.Client
-	defaultHeaders RequestHeaders
-	limiter        *rate.Limiter
+	client            *http.Client
+	defaultHeaders    RequestHeaders
+	limiter           *rate.Limiter
+	RetryMaxDuration  time.Duration
```

Comment:
It was because the client is in another package and couldn't access the fields. Unless we want to pass these into `NewClient` along the existing options?

Reply:
I'd duplicate the options then, so you have a `WithExponentialBackoff` option for the client too